src/test/java/marquez/dao/JobDAOTest.java

```diff
@@ -89,6 +97,39 @@ public void testInsert() {
     assertJobFieldsMatch(job, jobFound);
   }
 
+  @Test(expected = UnableToExecuteStatementException.class)
+  public void testInsert_Duplicate() {
+    UUID newNamespaceId = UUID.randomUUID();
+    insertNamespace(newNamespaceId, "newNsForDupTest", "Amaranta");
+    jobDAO.insert(job);
+
+    try {
+      Job jobWithDiffNsSameName =
+          new Job(
+              UUID.randomUUID(),
+              job.getName(),
+              "location",
+              newNamespaceId,
+              "desc",
+              Collections.<String>emptyList(),
+              Collections.<String>emptyList());
+      jobDAO.insert(jobWithDiffNsSameName);
+    } catch (UnableToExecuteStatementException e) {
+      fail("failed to insert a job with the same name into a different namespace");
```

Comment:
This is a little confusing. You are attempting to insert a job with the same name but a different NS than an existing job. This should work and your test should just pass. 

It also makes sense to me that you're catching the exception it can throw, and then marking the test as failed. It's not a necessary thing to do, since the exception being thrown is enough to fail the test, but some people like to do it that way and that's fine by me.

What's not clear to me is why you are expecting the UnableToExecuteStatementException. First off, is the test intending to demonstrate that this will always throw an exception? If so, then I don't think that's the agreed-upon behavior.
Secondly, since it's being expected, are you expecting it from the namespace creation request or first job creation request, since the second job creation attempt will catch it? If so, what would that demonstrate?